app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
+
+
+results = {}
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    def inner(*args: Any) -> Any:
+        global results
+        if (func, args) in results:
+            print("Getting from cache")
+            return results[(func, args)]
+        else:
+            print("Calculating new result")
+            results[(func, args)] = (func(*args))
+            return results[(func, args)]
```

Comment:
Rewrite with using one return statement

Reply:
Done. But now i have two opened pullrequests — this and next in line